### PR TITLE
Enable DataStorm multicast testing on macOS

### LINKED ISSUE #2929: DataStorm multicast testing hanging on GitHub macOS runners
I added multicast test configurations to DataStorm test cases in https://github.com/zeroc-ice/ice/pull/2927

But the multicast tests are hanging on macOS runners. See https://github.com/zeroc-ice/ice/actions/runs/11401253797/job/31723801953?pr=2927

They work fine on my local macOS box.

### PR DESCRIPTION
**Pull request by pepone**
Fixes #2929

### CODE REVIEW

File: scripts/DataStormUtil.py [hunk 1]
```diff
@@ -36,8 +36,9 @@ def __init__(self, instanceName=None, instance=None, *args, **kargs):
     def getEffectiveProps(self, current, props):
         props = DataStormProcess.getEffectiveProps(self, current, props)
         if ("DataStorm.Node.Multicast.Enabled", 1) in props.items():
-            props["DataStorm.Node.Multicast.Endpoints"] = f"udp -h 239.255.0.1 -p {current.driver.getTestPort(20)}"
-            props["DataStorm.Node.Multicast.PublishedHost"] = "239.255.0.1"
+            port = current.driver.getTestPort(20)
+            props["DataStorm.Node.Multicast.Endpoints"] = f"udp -h 239.255.0.1 -p {port}"
+            props["DataStorm.Node.Multicast.Proxy"] = f"DataStorm/Lookup -d:udp -h 239.255.0.1 --sourceAddress 127.0.0.1 -p {port}"
```
Comment (pepone) [inline scripts/DataStormUtil.py:41]: The multicast test on macOS GitHub runner fails when `--sourceAddress 127.0.0.1` is not set:

```
  local address = 192.168.64.22:58454
   remote address = 239.255.0.1:14120
   socket error: No route to host
```

Comment (externl) [reply scripts/DataStormUtil.py:41]: Any idea why?

Comment (pepone) [reply scripts/DataStormUtil.py:41]: It must be related to the runner configuration. It works well locally without this.

Comment (bernardnormier) [reply scripts/DataStormUtil.py:41]: We have a similar (related?) logic in the test scripts for multicast:
https://github.com/zeroc-ice/ice/blob/f604fb280cee43034a91f920e7023dd52ba82f6b/scripts/tests/IceDiscovery/simple.py#L20

**Review by externl — APPROVED** [review]

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**